backend/function-apps/api/trustees/trustees.function.test.ts

```diff
@@ -0,0 +1,110 @@
+import { HttpRequest, InvocationContext } from '@azure/functions';
+import handler from './trustees.function';
+import ContextCreator from '../../azure/application-context-creator';
+import { TrusteesController } from '../../../lib/controllers/trustees/trustees.controller';
+import { toAzureError, toAzureSuccess } from '../../azure/functions';
+import { ApplicationContext } from '../../../lib/adapters/types/basic';
+import { LoggerImpl } from '../../../lib/adapters/services/logger.service';
+import { CamsHttpResponseInit } from '../../../lib/adapters/utils/http-response';
+import { Trustee } from '../../../../common/src/cams/parties';
+
+// Mock dependencies
+jest.mock('../../azure/application-context-creator');
+jest.mock('../../../lib/controllers/trustees/trustees.controller');
+jest.mock('../../azure/functions');
+
+describe('Trustees Function', () => {
+  let mockRequest: HttpRequest;
+  let mockInvocationContext: InvocationContext;
+  let mockContext: ApplicationContext;
+  let mockController: jest.Mocked<TrusteesController>;
+  let mockLogger: Partial<LoggerImpl>;
+
+  beforeEach(() => {
+    mockRequest = {} as HttpRequest;
+    mockInvocationContext = {} as InvocationContext;
+    mockContext = {} as ApplicationContext;
+    mockLogger = {
+      info: jest.fn(),
+      error: jest.fn(),
+    };
+
+    mockController = {
+      handleRequest: jest.fn(),
+    } as unknown as jest.Mocked<TrusteesController>;
+
+    (ContextCreator.getLogger as jest.Mock).mockReturnValue(mockLogger);
+    (ContextCreator.applicationContextCreator as jest.Mock).mockResolvedValue(mockContext);
+    (ContextCreator.getApplicationContextSession as jest.Mock).mockResolvedValue({});
+    (TrusteesController as jest.MockedClass<typeof TrusteesController>).mockImplementation(
+      () => mockController,
+    );
+  });
+
+  afterEach(() => {
+    jest.restoreAllMocks();
+  });
+
+  test('should handle successful trustee request', async () => {
+    const mockResponse: Partial<CamsHttpResponseInit<Trustee>> = {
+      statusCode: 200,
+      body: { data: [] as unknown as Trustee },
+    };
+    const mockAzureResponse = { status: 200 };
+
+    mockController.handleRequest.mockResolvedValue(mockResponse);
+    (toAzureSuccess as jest.Mock).mockReturnValue(mockAzureResponse);
```

Comment:
This feels different than previous approaches to tests. For example, we don't typically mock `functions.ts` and its functions at all. Here we're mocking it on line 14 but then mocking a particular function in a way we don't typically mock anything. Do we like this approach?